pyccel/codegen/utilities.py

```diff
@@ -177,43 +179,52 @@ def manage_dependencies(pyccel_imports, compiler, pyccel_dirpath, mod_obj, langu
     # Iterate over the recognised_libs list and determine if the printer
     # requires a library to be included.
     for lib_name, stdlib in recognised_libs.items():
+        if stdlib is None:
+            continue
         if any(i == lib_name or i.startswith(f'{lib_name}/') for i in pyccel_imports):
             stdlib_obj = stdlib.install_to(pyccel_dirpath, installed_libs, verbose, compiler)
 
-            mod_obj.add_dependencies(stdlib_obj)
+            if isinstance(mod_obj, CompileObj):
+                mod_obj.add_dependencies(stdlib_obj)
 
             # stop after copying lib to __pyccel__ directory for
             # convert only
             if convert_only:
                 continue
 
-    for lib_obj in installed_libs.values():
-        # get the include folder path and library files
-        recompile_object(lib_obj,
-                         compiler = compiler,
-                         language = language,
-                         verbose  = verbose)
+    if not convert_only:
+        lib_compile_objs = [lib_obj for key, lib_obj in installed_libs.items() if key != 'gFTL_extensions']
+        lib_compile_objs.extend(installed_libs.get('gFTL_extensions', {}).values())
+        for lib_obj in lib_compile_objs:
+            # get the include folder path and library files
+            recompile_object(lib_obj,
+                             compiler = compiler,
+                             language = language,
+                             verbose  = verbose)
 
     # Iterate over the imports and determine if the printer
     # requires an extension module to be generated
     for key, import_node in pyccel_imports.items():
         deps = generate_extension_modules(key, import_node, pyccel_dirpath,
                                           compiler     = compiler,
-                                          include     = mod_obj.include,
-                                          libs         = mod_obj.libs,
-                                          libdir      = mod_obj.libdir,
+                                          include      = getattr(mod_obj, 'include', ()),
+                                          libs         = getattr(mod_obj, 'libs', ()),
+                                          libdir       = getattr(mod_obj, 'libdir', ()),
```

Comment:
Here you are treating the case where `mod_obj` is not an instance of `CompileObj`, right?